src/jsonUtils/computeNode.js

```diff
@@ -0,0 +1,445 @@
+      if (isPercent(style.marginBottom)) {
+        yogaNode.setMarginPercent(yoga.EDGE_BOTTOM, style.marginBottom);
+      } else if (isAuto(style.marginBottom)) {
+        yogaNode.setMarginAuto(yoga.EDGE_BOTTOM);
+      } else {
+        yogaNode.setMargin(yoga.EDGE_BOTTOM, style.marginBottom);
+      }
+    }
+    if (!isNullOrUndefined(style.marginLeft)) {
+      if (isPercent(style.marginLeft)) {
+        yogaNode.setMarginPercent(yoga.EDGE_LEFT, style.marginLeft);
+      } else if (isAuto(style.marginLeft)) {
+        yogaNode.setMarginAuto(yoga.EDGE_LEFT);
+      } else {
+        yogaNode.setMargin(yoga.EDGE_LEFT, style.marginLeft);
+      }
+    }
+    if (!isNullOrUndefined(style.marginRight)) {
+      if (isPercent(style.marginRight)) {
+        yogaNode.setMarginPercent(yoga.EDGE_RIGHT, style.marginRight);
+      } else if (isAuto(style.marginRight)) {
+        yogaNode.setMarginAuto(yoga.EDGE_RIGHT);
+      } else {
+        yogaNode.setMargin(yoga.EDGE_RIGHT, style.marginRight);
+      }
+    }
+    if (!isNullOrUndefined(style.marginVertical)) {
+      if (isPercent(style.marginVertical)) {
+        yogaNode.setMarginPercent(yoga.EDGE_VERTICAL, style.marginVertical);
+      } else if (isAuto(style.marginVertical)) {
+        yogaNode.setMarginAuto(yoga.EDGE_VERTICAL);
+      } else {
+        yogaNode.setMargin(yoga.EDGE_VERTICAL, style.marginVertical);
+      }
+    }
+    if (!isNullOrUndefined(style.marginHorizontal)) {
+      if (isPercent(style.marginHorizontal)) {
+        yogaNode.setMarginPercent(yoga.EDGE_HORIZONTAL, style.marginHorizontal);
+      } else if (isAuto(style.marginHorizontal)) {
+        yogaNode.setMarginAuto(yoga.EDGE_HORIZONTAL);
+      } else {
+        yogaNode.setMargin(yoga.EDGE_HORIZONTAL, style.marginHorizontal);
+      }
+    }
+    if (!isNullOrUndefined(style.margin)) {
+      if (isPercent(style.margin)) {
+        yogaNode.setMarginPercent(yoga.EDGE_ALL, style.margin);
+      } else if (isAuto(style.margin)) {
+        yogaNode.setMarginAuto(yoga.EDGE_ALL);
+      } else {
+        yogaNode.setMargin(yoga.EDGE_ALL, style.margin);
+      }
+    }
+
+    // Padding
+    if (!isNullOrUndefined(style.paddingTop)) {
+      if (isPercent(style.paddingTop)) {
+        yogaNode.setPaddingPercent(yoga.EDGE_TOP, style.paddingTop);
+      } else {
+        yogaNode.setPadding(yoga.EDGE_TOP, style.paddingTop);
+      }
+    }
+    if (!isNullOrUndefined(style.paddingBottom)) {
+      if (isPercent(style.paddingBottom)) {
+        yogaNode.setPaddingPercent(yoga.EDGE_BOTTOM, style.paddingBottom);
+      } else {
+        yogaNode.setPadding(yoga.EDGE_BOTTOM, style.paddingBottom);
+      }
+    }
+    if (!isNullOrUndefined(style.paddingLeft)) {
+      if (isPercent(style.paddingLeft)) {
+        yogaNode.setPaddingPercent(yoga.EDGE_LEFT, style.paddingLeft);
+      } else {
+        yogaNode.setPadding(yoga.EDGE_LEFT, style.paddingLeft);
+      }
+    }
+    if (!isNullOrUndefined(style.paddingRight)) {
```

Comment:
Would it make sense to extract these `setPadding` into a function that takes the edge and a value to remove duplication?

Reply:
For now, I prefer being a bit more verbose here to make the code more easily understandable. I messed with trying to dynamically set properties using maps, but it made the code much harder to figure out :( . Definitely open to ideas, but maybe there can be a separate PR to address the verboseness?

Reply:
👍